Bindings/Python/CMakeLists.txt

```diff
@@ -5,5 +5,10 @@ find_package(PythonInterp REQUIRED)
 include_directories(${PYTHON_INCLUDE_DIRS})
 
 add_subdirectory(pybind11)
-add_compile_options("/std:c++17" "/permissive-")
+if (WIN32)
+    add_compile_options("/std:c++17" "/permissive-")
+endif (WIN32)
+if (UNIX)
+    add_compile_options("-std=c++17" "-fpermissive")
```

Comment:
In windows we are specifying "permissive-" to disable the permissive setting.  It looks like for Unix we are making things permissive.  I think we probably just need to remove that option for Linux/Unix builds.  I also don't think clang supports it.

Reply:
I should have specified that I'm using GCC, where it is supposed to
> Downgrade some diagnostics about nonconformant code from errors to warnings. Thus, using -fpermissive allows some nonconforming code to compile.  
https://gcc.gnu.org/onlinedocs/gcc-4.8.0/gcc/C_002b_002b-Dialect-Options.html#index-fpermissive-154